cf/test/test_Query.py

```diff
@@ -602,6 +618,134 @@ def test_Query__or__(self):
         self.assertEqual(s.value, 9)
         self.assertFalse(s == 8)
 
+        q = cf.set(cf.Data([1])) | cf.set(cf.Data([1]))
+        self.assertEqual(q.value, cf.Data([1]))
+
+        q = cf.set([1]) | cf.set(cf.Data([1]))
+        self.assertEqual(q.value, [1])
+
+        q = cf.set([1, 2]) | cf.set(cf.Data([1]))
+        self.assertIsNone(q._value)
+
+        q = cf.set([1]) | cf.set(cf.Data([1, 2]))
+        self.assertIsNone(q._value)
+
+        q = cf.set(cf.Data([1, 2, 3])) | cf.set(cf.Data([1, 2]))
+        self.assertIsNone(q._value)
+
+    def test_Query__dask_tokenize__(self):
+        for q in (
+            cf.eq(9),
+            cf.lt(9, "km"),
+            cf.wi(2, 5, "km"),
+            cf.set([1, 2, 3]),
+            cf.set([1, 2, 3], "km"),
+            cf.ge(7, attr="year"),
+            cf.ge(7, attr="upper_bounds.month", units="K"),
+            cf.eq(8) & cf.le(7, "km"),
+            cf.wo(2, 5, attr="day") | cf.set(cf.Data([1, 2], "km")),
+            cf.eq(8) | cf.lt(9) & cf.ge(10),
+            cf.isclose(1, "days", rtol=10, atol=99),
+        ):
+            self.assertEqual(tokenize(q), tokenize(q.copy()))
+
+        self.assertEqual(tokenize(cf.eq(9, "km")), tokenize(cf.eq(9, "1000m")))
+        self.assertNotEqual(
+            tokenize(cf.eq(9, "km")), tokenize(cf.eq(9000, "m"))
+        )
```

Comment:
Thinking ....

Reply:
Discussed off-line:

* Equivalent queries (or time durations) with different units (e.g. `cf.eq(9, "km")` and `cf.eq(9000, "m")`) do not affect the aggregation, because whichever one is first in the sequence is always used, so the token that appears in the aggregation signature will always be the same.

* It is not clear at this time how you would come up with a canonical tokenisation, as each tokenisation is done independently - how would no to convert its units and data to a canonical unit, and what would that unit be?

In the light of this, it makes sense to leave the tokenisation strategy as is, and worry about it if ever a use case arises that can't be handled by the code calling the tokenisations.